src/core/crypto/multiKey.ts

```diff
@@ -1,12 +1,31 @@
+import { AuthenticationKey } from "../authenticationKey";
 import { Hex } from "../hex";
-import { HexInput } from "../../types";
+import { HexInput, SigningScheme as AuthenticationKeyScheme } from "../../types";
 import { Deserializer } from "../../bcs/deserializer";
-import { Serializer } from "../../bcs/serializer";
-import { AnyPublicKey } from "./anyPublicKey";
-import { AnySignature } from "./anySignature";
-import { PublicKey } from "./asymmetricCrypto";
+import { Serializable, Serializer } from "../../bcs/serializer";
+import { Ed25519PublicKey, Ed25519Signature } from "./ed25519";
+import { Secp256k1PublicKey, Secp256k1Signature } from "./secp256k1";
+import { AnyPublicKey, AnySignature } from "./singleKey";
+
+type PublicKeyInput = Ed25519PublicKey | Secp256k1PublicKey | AnyPublicKey;
```

Comment:
I feel this type repeats itself in a couple of places?

Reply:
This is slightly different. Namely you can't use a Multikey as input oif a MultiKey.
Or I guess you could but it's easier not to support it for now 😆